### PR TITLE
Migrate from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by qmfrederik**
travis-ci.org has finally shut down; this PR simply migrates the Travis CI script to GitHub actions.

### CODE REVIEW

**Comment by tmds** [comment]
@qmfrederik thanks for migrating this. I have the same comments as in https://github.com/tmds/Tmds.DBus/pull/105.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - run: |
+        pushd test/Tmds.LibC.Tests && dotnet test && popd
+        dotnet pack src/Tmds.LibC --configuration Release --version-suffix "$(date +"%y%m%d")-$TRAVIS_BUILD_NUMBER" --output .
+        
+    - uses: actions/upload-artifact@v2
```
Comment (tmds) [inline .github/workflows/build.yaml:18]: Can you remove this step?

Comment (qmfrederik) [reply .github/workflows/build.yaml]: I removed this step, although I found it useful to be able to retrieve the NuGet packages from PR builds.

**Comment by tmds** [comment]
This is also working fine!

**Pull request merged**